### PR TITLE
Enable goconst

### PR DESCRIPTION
**Pull request by daxmc99**
_Provide a description of what has been changed_

Part of https://github.com/kedacore/keda/issues/1158

Enable `goconst`
Planning to enable `gocritic` next

Happy Hacktober 🎃

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
~~- [ ] Tests have been added~~ N/A
~~- [ ] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs~~ N/A
- [ ] Changelog has been updated

### CODE REVIEW

File: pkg/scalers/metrics_api_scaler.go [hunk 1]
```diff
@@ -134,8 +134,8 @@ func parseMetricsAPIMetadata(config *ScalerConfig) (*metricsAPIScalerMetadata, e
 		meta.method = "header"
 		meta.enableAPIKeyAuth = true
 
-		if config.TriggerMetadata["method"] == "query" {
-			meta.method = "query"
+		if config.TriggerMetadata["method"] == promQuery {
```
Comment (zroubalik) [inline pkg/scalers/metrics_api_scaler.go:137]: It is not ideal to reference const from another scaler, we should introduce a one specific for this scaler instead

**Review by zroubalik — CHANGES_REQUESTED** [review]

**Comment by zroubalik** [comment]
@daxmc99 hi, any update on this please?

**Comment by Shubham82** [comment]
NOTE: I am working on this and i have created a new PR for this. #1341

**Comment by daxmc99** [comment]
Thanks for picking this up @Shubham82 
Sorry for the delay @zroubalik

**Comment by zroubalik** [comment]
@daxmc99 no worries :)

**Comment by Shubham82** [comment]
Hi @daxmc99  
Are you planning to enable gocritic or not.

**Pull request closed without merging**